modules/tensor_mechanics/src/materials/crystal_plasticity/CrystalPlasticityTwinningKalidindiUpdate.C

```diff
@@ -0,0 +1,339 @@
+      "non_coplanar_coefficient_twin_hardening>=0",
+      "The factor to apply to the hardening of non-coplanar twin systems "
+      "strength as a function of the volume fraction of twins");
+  params.addRangeCheckedParam<Real>(
+      "coplanar_coefficient_twin_hardening",
+      800.0,
+      "coplanar_coefficient_twin_hardening>=0",
+      "Hardening coefficient for coplanar twin systems strength as a function of "
+      "the volume fraction of twins");
+  params.addRangeCheckedParam<Real>(
+      "non_coplanar_twin_hardening_exponent",
+      0.05,
+      "non_coplanar_twin_hardening_exponent>=0",
+      "Parameter used to increase the hardening of non-coplanar twin systems such that the "
+      "propagation of twins on conplanar systems is favored at early deformation stages.");
+  params.addRangeCheckedParam<Real>(
+      "upper_limit_twin_volume_fraction",
+      0.8,
+      "upper_limit_twin_volume_fraction>0&upper_limit_twin_volume_fraction<=1",
+      "The maximumum amount of twinning volume fraction allowed");
+
+  return params;
+}
+
+CrystalPlasticityTwinningKalidindiUpdate::CrystalPlasticityTwinningKalidindiUpdate(
+    const InputParameters & parameters)
+  : CrystalPlasticityStressUpdateBase(parameters),
+
+    _total_twin_volume_fraction(declareProperty<Real>(_base_name + "total_volume_fraction_twins")),
+    _total_twin_volume_fraction_old(
+        getMaterialPropertyOld<Real>(_base_name + "total_volume_fraction_twins")),
+    _initial_total_twin_volume_fraction(getParam<Real>("initial_total_twin_volume_fraction")),
+    _twin_volume_fraction(
+        declareProperty<std::vector<Real>>(_base_name + "twin_system_volume_fraction")),
+    _twin_volume_fraction_old(
+        getMaterialPropertyOld<std::vector<Real>>(_base_name + "twin_system_volume_fraction")),
+    _twin_volume_fraction_increment(
+        declareProperty<std::vector<Real>>(_base_name + "twin_system_volume_fraction_increment")),
+    _reference_strain_rate(getParam<Real>("twin_reference_strain_rate")),
+    _rate_sensitivity_exponent(getParam<Real>("twin_strain_rate_sensitivity_exponent")),
+    _characteristic_twin_shear(getParam<Real>("characteristic_twin_shear")),
+    _twin_initial_lattice_friction(getParam<Real>("initial_twin_lattice_friction")),
+    _non_coplanar_coefficient_twin_hardening(
+        getParam<Real>("non_coplanar_coefficient_twin_hardening")),
+    _coplanar_coefficient_twin_hardening(getParam<Real>("coplanar_coefficient_twin_hardening")),
+    _noncoplanar_exponent(getParam<Real>("non_coplanar_twin_hardening_exponent")),
+    _limit_twin_volume_fraction(getParam<Real>("upper_limit_twin_volume_fraction"))
+{
+  // resize local caching vectors used for substepping
+  _previous_substep_twin_resistance.resize(_number_slip_systems);
+  _previous_substep_twin_volume_fraction.resize(_number_slip_systems);
+  _twin_resistance_before_update.resize(_number_slip_systems);
+  _twin_volume_fraction_before_update.resize(_number_slip_systems);
```

Comment:
those should be initialized to that size in the initializer list

Reply:
I hope I made this change correctly, but if not please point me in the right direction